### PR TITLE
Flex Task

### PR DESCRIPTION
**Pull request by pranotoism**
### Detailed description of changes
#196/FAPagination
#190/FAFormControlLabel (wrapper)
#189/FAFormControl (wrapper)
#192/FAInput
#195/FAOutlinedInput ( also used as select)
...

### Recommendations on how to test the changes

...

### Checklist for developer

- [x] The code builds clean without any errors or warnings
- [x] The code does not contain commented out code
- [x] The code does not log anything to console
- [x] I have added unit test(s) to cover new code and successfully executed ran it
- [x] I have thoroughly tested the new code and any adjacent features it may affect

### Link to related issue(s)

...

#### Before

...

#### After
Screen Shot 2022-02-14 at 10 18 20 AM

Screen Shot 2022-02-14 at 10 54 15 AM
red helper text for error only

Screen Shot 2022-02-14 at 10 22 17 AM

...

### CODE REVIEW

File: src/Components/Atoms/FAPagination/index.tsx [hunk 1]
```diff
@@ -1,12 +1,45 @@
...
+    },
+    "& li button": {
+      fontSize: "16px",
+    },
+  },
+  small: {
+    "& li .Mui-selected": {
+      width: "32px",
+      height: "32px",
+      // height: "100%",
```
Comment (ardiwine) [inline src/Components/Atoms/FAPagination/index.tsx:22]: remove commented code

**Review by ardiwine — CHANGES_REQUESTED** [review]

File: src/Components/Atoms/FAPagination/index.tsx [hunk 2]
```diff
@@ -1,12 +1,44 @@
...
+    "& li .Mui-selected": {
+      width: "32px",
+      height: "32px",
+      borderRadius: "20px",
+    },
+    "& li button": {
+      fontSize: "14px",
+    },
+  },
+}));
```
Comment (gilangdarmawan) [inline src/Components/Atoms/FAPagination/index.tsx:28]: sebaiknya atur style di baseTheme.

File: src/Components/Atoms/FAPagination/index.tsx [hunk 3]
```diff
@@ -1,12 +1,44 @@
...
 
 export const FAPagination = memo((props: IFAPagination) => {
-  const { testID, count, ...materialUIProps } = props;
+  const { testID, count, size, ...materialUIProps } = props;
+  const classes = useStyles();
   return (
     <Stack spacing={2}>
-      <Pagination count={count} data-testid={testID} {...materialUIProps} />
+      <Pagination
+        className={size === "small" ? classes.small : classes.large}
```
Comment (gilangdarmawan) [inline src/Components/Atoms/FAPagination/index.tsx:36]: Jika sudah ditambahkan style nya di baseTheme, className bisa di remove

File: src/Components/Atoms/FATextField/index.tsx [hunk 4]
```diff
@@ -1,6 +1,17 @@
 import React, { forwardRef, memo } from "react";
 import { TextField, InputAdornment } from "@mui/material";
 import { IInput } from "./Input.interface";
+import { makeStyles } from "@mui/styles";
+
+const useStyles = makeStyles(() => ({
```
Comment (gilangdarmawan) [inline src/Components/Atoms/FATextField/index.tsx:6]: perlu konfirmasi ulang ke @muhammad-farhan64 untuk FATextField

File: src/Definitions/Theme/Theme.tsx [hunk 5]
```diff
@@ -74,14 +74,33 @@ export const baseTheme = (props: ThemeProps) =>
               fontWeight: 600,
               letterSpacing: "0.2px",
             },
+            ".MuiOutlinedInput-input": {
```
Comment (gilangdarmawan) [inline src/Definitions/Theme/Theme.tsx:77]: Perlu discuss dan pairing bareng terkait best practices untuk overrides style di baseTheme

**Review by gilangdarmawan — CHANGES_REQUESTED** [review]

File: src/Definitions/Theme/Theme.tsx (hunk 5 above)
Comment (ardiwine) [reply src/Definitions/Theme/Theme.tsx:77]: @RikaSuri tolong di bantu

File: src/Components/Atoms/FATextField/index.tsx (hunk 4 above)
Comment (ardiwine) [reply src/Components/Atoms/FATextField/index.tsx:6]: @muhammad-farhan64 tolong di respond ya thank you.

File: src/Definitions/Theme/Theme.tsx [hunk 6]
```diff
@@ -50,6 +50,25 @@ declare module "@mui/material/Typography" {
 export const baseTheme = (props: ThemeProps) =>
   createTheme({
     components: {
+      MuiPagination: {
+        styleOverrides: {
+          root: {
+            ".MuiPaginationItem-sizeSmall.Mui-selected": {
```
Comment (gilangdarmawan) [inline src/Definitions/Theme/Theme.tsx:56]: ini bisa langsung override component MuiPaginationItem. bisa cek dokumentasi untuk best practice nya [disini](https://mui.com/customization/theme-components/#global-style-overrides)

**Review by gilangdarmawan — CHANGES_REQUESTED** [review]

**Review by gilangdarmawan — APPROVED** [review]

**Review by ardiwine — APPROVED** [review]

**Pull request merged**